### ISSUE TITLE
NOAUTH Authentication required

### ISSUE THREAD

**Issue by Somebi** [twitter/twemproxy#491]
nc_redis.c:2936 SELECT 10 failed on redis-pool | my_ip_here:6379: NOAUTH Authentication required.

nutcracker -d -c /var/local/nutcracker.yml -a 0.0.0.0 -s 20100 -v 11 -o /var/log/nutcracker.log

redis-pool:
  listen: 127.0.0.1:20001
  distribution: random
  redis: true
  servers:
   - my_ip_here:6379:1
  redis_db: 10
  redis_auth: my_password_here

With preconnect: true throwing right away.

Select command is not supported, so i fearing that it's not selecting 10 redis_db and using 0 db as default one.

Could you please look into it?

**Comment by Somebi** [comment]
Yap it's not selecting required 10 database. It's selecting default 0 database.

**Comment by Somebi** [comment]
redis_auth should have password in there or just a boolean? redis_auth: true or redis_auth: password.
Password should have this format "password" or just clean password?
Something is wrong there definitely.

**Comment by Somebi** [comment]
Just tried to compile from master, by cloning. Same results. preconnect: true with password and redis_db: 10 for example.

`nc_redis.c:2948 SELECT 10 failed on redis-pool | my_ip:6379: NOAUTH Authentication required.`

Worth look into that problem i think.

**Comment by charsyam** [comment]
@Somebi Your client should send auth command to twemproxy.
this doesn't support to skip enter password for client.

**Comment by Somebi** [comment]
Yes, but it doesn't allow to select database from client. But to select it from twemproxy I need provide auth password... How do i do that? O_o

**Comment by therealbill** [comment]
When a Redis instance is configured for authentication, you *must* send the auth command before it will let you do anything other than disconnect.  If you have to use the select command, you can't use Twemproxy as it isn't a supported command.

**Comment by Somebi** [comment]
But in twemproxy there is configuration option for that...

`redis_db: The DB number to use on the pool servers. Defaults to 0. Note: Twemproxy will always present itself to clients as DB 0.`

What this option for then?

`redis_auth: Authenticate to the Redis server on connect.`

Should be removed from documentation then...

If you can't use these options together `**preconnect + redis_db + redis_auth**`, then it's should be documented as well.

**Comment by Somebi** [comment]
Guess i'll have to use ssh tunnel to redis server to use twemproxy without password...

**Comment by charsyam** [comment]
@Somebi 
also. You can use this patch, https://github.com/twitter/twemproxy/pull/458
